reference/arch/online_arch.md

```diff
@@ -0,0 +1,45 @@
+# 在线模块架构
+
+## 1. 概览
+
+OpenMLDB 的在线架构的主要模块包括 Apache ZooKeeper, Nameserver 以及 Tablets（进一步包含了SQL Engine 和 Storage Engine）。如下图显示了这些模块之间的相互关系。其中 Tablets 是整个 OpenMLDB 存储和计算的核心模块，也是消耗资源做多的模块；ZooKeeper 和 Nameserver 主要用于辅助功能，如元数据的管理和高可用等。本文以下将会详细介绍各个模块的作用。
+![image-20220316160612968](images/architecture.png)
+
+
+## 2. Apache ZooKeeper
+OpenMLDB 依赖 ZooKeeper 做服务发现和元数据存储和管理功能。ZooKeeper 和 OpenMLDB SDK，tables, namesever 之间都会存在交互，用于分发和更新元数据。
+
+## 3. Nameserver
+Nameserver 主要用来做 tablet 管理以及故障转移（failover）。当一个 Tablet 节点宕机后，nameserver 就会触发一系列任务来执行故障转移，当节点恢复后会重新把数据加载到该节点中。故障转移和数据恢复是以分片（partition）为单位的，Storage Engine 部分有分片的详细介绍。

```

Comment:
Unify the name, Nameserver/nameserver, Tablet/tablet

Reply:
right, this one is missing, only the first word of a sentence should be capitalized.